frontend/packages/console-shared/src/components/status/status.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import {
+  HourglassStartIcon,
+  HourglassHalfIcon,
+  SyncIcon,
+  BanIcon,
+  WarningTriangleIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import {
+  StatusIconAndText,
+  ProgressStatus,
+  ErrorStatus,
+  SuccessStatus,
+  InfoStatus,
+} from './status-components';
+
+export type StatusProps = {
+  status?: string;
+};
+
+export const Status: React.FC<StatusProps> = ({ status, children }) => {
+  switch (status) {
+    case 'New':
+      return <StatusIconAndText icon={<HourglassStartIcon />} title={status} />;
+
+    case 'Pending':
+      return <StatusIconAndText icon={<HourglassHalfIcon />} title={status} />;
+
+    case 'ContainerCreating':
+      return <ProgressStatus title={status} />;
+
+    case 'In Progress':
+    case 'Running':
+    case 'Updating':
+    case 'Upgrading':
+      return <StatusIconAndText icon={<SyncIcon />} title={status} />;
+
+    case 'Cancelled':
+    case 'Expired':
+    case 'Not Ready':
+    case 'Terminating':
+      return <StatusIconAndText icon={<BanIcon />} title={status} />;
+
+    case 'Warning':
+      return <StatusIconAndText icon={<WarningTriangleIcon />} title={status} />;
+
+    case 'ContainerCannotRun':
+    case 'CrashLoopBackOff':
+    case 'Critical':
+    case 'Error':
+    case 'Failed':
+    case 'InstallCheckFailed':
+    case 'Lost':
+    case 'Rejected':
+      return <ErrorStatus title={status}>{children}</ErrorStatus>;
+
+    case 'Accepted':
+    case 'Active':
+    case 'Bound':
+    case 'Complete':
+    case 'Completed':
+    case 'Enabled':
+    case 'Ready':
+    case 'Up to date':
```

Comment:
Nit: this is inconsistent with other status values where words are always capitalized, but I guess we want to keep it this way for now.